### PR TITLE
Moped Database Replication Tool

### PR DESCRIPTION
**Pull request by frankhereford**
## Associated issues

_none_

## Purpose

This PR contains a small docker-compose stack which provides the tools to automate the cloning the moped database from a remote postgres server to your local one, including the `hdb` schemas which configure `graphql-engine`. It's a tool which becomes useful when you need to repeatedly get data which has been placed on the read replica or the heroku test database instance, perhaps by the lambda functions executed as a result of changes to the Moped data.

Initially, I did not intend to try to PR this into the toolbox and just keep it around as a personal tool, but if others on the dev team would like to see this included, I hope this PR provides the information needed. If this isn't a good fit for the repo, that's fine with me, of course!

## Considerations (ARM64 only for now)

This PR updates the image used locally to be based on PG V13 instead of V12. This is to get local up to the release of the Heroku deployment. Production remains on V12.

I will make this same change to V13 for Intel users in a few weeks. I will create an issue for the correct sprint if this PR ends up being merged. Letting ARM64 run it a few weeks in advance serves as some sort of testing before upgrading it for the rest of the devs. So far, for the last week or so, it's worked for me with zero issues at all.

## Testing

There is not a URL for testing this PR, as this is a CLI tool aimed at automating DB replication for the moped app from a remote database to a local one. I have tested this tool pulling from both our `test` heroku DB and our production DB via the read replica.

**Steps to test:**
1) Create a `.env` file that contains the credentials for the database you wish to pull data from. There is a `env_template` file to demonstrate the form of this environment file.
1) Spin up your local database. Consider the version you're running in relation to the version of the source data.
1) Change to the `moped-toolbox/pg_replcate_to_local` directory.
1) Execute `scripts/reload_from_remote_db.sh`

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [x] ~Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#gid=776973707)~

### CODE REVIEW

**Comment by frankhereford** [comment]
FWIW, I have used this system to replicate the database from both the heroku test database, but also now from the read-replica. This makes for a pretty effective way to get the latest data from production if needed.

File: moped-toolbox/pg_replicate_to_local/scripts/pull_down_remote_db.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+echo "Downloading remote DB"
+pg_dump --verbose --no-owner > /root/workspace/db.sql
```
Comment (mateoclarke) [inline moped-toolbox/pg_replicate_to_local/scripts/pull_down_remote_db.sh:4]: I got this error:
```
➜  pg_replicate_to_local git:(flh_database_replication_tool) scripts/pull_down_remote_db.sh
Downloading remote DB
scripts/pull_down_remote_db.sh: line 4: /root/workspace/db.sql: No such file or directory
```

**Review by mateoclarke — CHANGES_REQUESTED** [review]

**Comment by frankhereford** [comment]
@chiaberry @mateoclarke 

I have updated the comment to include the correct script to run: `scripts/reload_from_remote_db.sh`.

I had it right in the `README.md`, but I had goofed it up in the PR comment. Thank you both for catching that!

File: moped-toolbox/pg_replicate_to_local/scripts/pull_down_remote_db.sh (hunk 1 above)
Comment (frankhereford) [reply moped-toolbox/pg_replicate_to_local/scripts/pull_down_remote_db.sh:4]: Thanks! I had a typo in my instructions in the PR, thank you for your help!

**Comment by chiaberry** [comment]
Ran the script, did get a couple of errors at the end: 
```
utilities_1  | pg_dump: saving encoding = UTF8
utilities_1  | pg_dump: saving standard_conforming_strings = on
utilities_1  | pg_dump: saving search_path = 
utilities_1  | pg_dump: error: query failed: ERROR:  permission denied for sequence remote_schemas_id_seq
utilities_1  | pg_dump: error: query was: SELECT last_value, is_called FROM hdb_catalog.remote_schemas_id_seq
pg_replicate_to_local_utilities_1 exited with code 1
DROP DATABASE
CREATE DATABASE
CREATE EXTENSION
moped-database_moped-pgsql_1 is up-to-date
Starting moped-database_hasura_1 ... done
```
This is what I get when I check local moped:
Screen Shot 2022-03-09 at 3 53 11 PM
and the hasura console is empty

**Comment by frankhereford** [comment]
@chiaberry and I are going to meet this afternoon to investigate this error. Thanks for finding it!

**Comment by frankhereford** [comment]
@chiaberry and I met today to review an error she is encountering when replicating data. She may have a screenshot to post to demonstrate what is happening. We agreed that this will be easier to work on when we're both at the office. We'll pick this back up and move it forward in a couple of weeks.

Additionally, I am going to fire up a X86 Mac I have at the house to attempt to replicate the issue she is having to work up a solution as well. 

Until the issue is replicated and resolved, I am going to move this back into WIP.

**Comment by chiaberry** [comment]
Screen Shot 2022-03-11 at 9 34 59 AM

**Comment by frankhereford** [comment]
I'm going to close this for now. It's been a loooong time since I've looked at it, and there are plenty of other things that are burning brighter than this on the list. 🔥 Might pick it up again if it becomes more needed.

**Pull request closed without merging**